packages/@aws-cdk/aws-apprunner-alpha/lib/auto-scaling-configuration.ts

```diff
@@ -0,0 +1,198 @@
+   *
+   * @default - a name generated by CloudFormation
+   */
+  readonly autoScalingConfigurationName?: string;
+
+  /**
+   * The maximum number of concurrent requests that an instance processes.
+   * If the number of concurrent requests exceeds this limit, App Runner scales the service up.
+   *
+   * Must be between 1 and 200.
+   *
+   * @default 100
+   */
+  readonly maxConcurrency?: number;
+
+  /**
+   * The maximum number of instances that a service scales up to.
+   * At most maxSize instances actively serve traffic for your service.
+   *
+   * Must be between 1 and 25.
+   *
+   * @default 25
+   */
+  readonly maxSize?: number;
+
+  /**
+   * The minimum number of instances that App Runner provisions for a service.
+   * The service always has at least minSize provisioned instances.
+   *
+   *
+   * Must be between 1 and 25.
+   *
+   * @default 1
+   */
+  readonly minSize?: number;
+}
+
+/**
+ * Attributes for the App Runner Auto Scaling Configuration.
+ */
+export interface AutoScalingConfigurationAttributes {
+  /**
+   * The name of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The ARN of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * Represents the App Runner Auto Scaling Configuration.
+ */
+export interface IAutoScalingConfiguration extends cdk.IResource {
+  /**
+   * The ARN of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The Name of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * The App Runner Auto Scaling Configuration.
+ *
+ * @resource AWS::AppRunner::AutoScalingConfiguration
+ */
+export class AutoScalingConfiguration extends cdk.Resource implements IAutoScalingConfiguration {
+  /**
+   * Imports an App Runner Auto Scaling Configuration from attributes
+   */
+  public static fromAutoScalingConfigurationAttributes(scope: Construct, id: string,
+    attrs: AutoScalingConfigurationAttributes): IAutoScalingConfiguration {
+    const autoScalingConfigurationArn = attrs.autoScalingConfigurationArn;
+    const autoScalingConfigurationName = attrs.autoScalingConfigurationName;
+    const autoScalingConfigurationRevision = attrs.autoScalingConfigurationRevision;
+
+    class Import extends cdk.Resource {
+      public readonly autoScalingConfigurationArn = autoScalingConfigurationArn
+      public readonly autoScalingConfigurationName = autoScalingConfigurationName
+      public readonly autoScalingConfigurationRevision = autoScalingConfigurationRevision
+    }
+
+    return new Import(scope, id);
+  }
+
```

Comment:
If we just know the Name and Revision, we probably can formate the Arn string.

How about just have Name and Revision in the interface:

```ts
export interface AutoScalingConfigurationAttributes {
  /**
   * The name of the Auto Scaling Configuration.
   */
  readonly autoScalingConfigurationName: string;

  /**
   * The revision of the Auto Scaling Configuration.
   */
  readonly autoScalingConfigurationRevision: number;
}
```

And in the fromXxx method:

```ts
public static fromAutoScalingConfigurationAttributes(scope: Construct, id: string,
    attrs: AutoScalingConfigurationAttributes): IAutoScalingConfiguration {

    class Import extends cdk.Resource implements IAutoScalingConfiguration {
       public readonly autoScalingConfigurationName = attrs.autoScalingConfigurationName;
       public readonly autoScalingConfigurationRevision = attrs.autoScalingConfigurationRevision;
       public readonly autoScalingConfigurationArn = Stack.of(this).formatArn({
           resource: 'autoscalingconfiguration',
           service: 'apprunner',
           resourceName: `${attrs.autoScalingConfigurationName}/${attrs.autoScalingConfigurationRevision}`,
      })
    }
    
    return new Import(scope, id);
};
```

If we can generate the Arn from given Name and Revision then Arn would not be required in the Attributes?

Reply:
Thank you. You are certainly right. I have removed arn from the attributes and made the correction.